packages/itwinui-react/src/core/Tabs/Tabs.test.tsx

```diff
@@ -237,79 +282,185 @@ it.each(['horizontal', 'vertical'] as const)(
 
     // 0 -> 1
-    fireEvent.keyDown(tablist, { key: nextTabKey });
-    expect(mockOnTabSelected).toBeCalledWith(1);
+    fireEvent.keyDown(tabs[0], { key: nextTabKey });
+    expect(mockonActivated).toBeCalledWith('tab1');
     expect(document.activeElement).toBe(tabs[1]);
 
     // 1 -> 2
-    fireEvent.keyDown(tablist, { key: nextTabKey });
-    expect(mockOnTabSelected).toBeCalledWith(2);
+    fireEvent.keyDown(tabs[1], { key: nextTabKey });
+    expect(mockonActivated).toBeCalledWith('tab2');
     expect(document.activeElement).toBe(tabs[2]);
 
     // 2 -> 0
-    fireEvent.keyDown(tablist, { key: nextTabKey });
-    expect(mockOnTabSelected).toBeCalledWith(0);
+    fireEvent.keyDown(tabs[2], { key: nextTabKey });
+    expect(mockonActivated).toBeCalledWith('tab0');
     expect(document.activeElement).toBe(tabs[0]);
 
     // 0 -> 2
-    fireEvent.keyDown(tablist, { key: previousTabKey });
-    expect(mockOnTabSelected).toBeCalledWith(2);
+    fireEvent.keyDown(tabs[0], { key: previousTabKey });
+    expect(mockonActivated).toBeCalledWith('tab2');
     expect(document.activeElement).toBe(tabs[2]);
 
     // 2 -> 1
-    fireEvent.keyDown(tablist, { key: previousTabKey });
-    expect(mockOnTabSelected).toBeCalledWith(1);
+    fireEvent.keyDown(tabs[2], { key: previousTabKey });
+    expect(mockonActivated).toBeCalledWith('tab1');
     expect(document.activeElement).toBe(tabs[1]);
   },
 );
 
-it('should handle keypresses when focusActivationMode is manual', () => {
-  const mockOnTabSelected = jest.fn();
-  const { container } = renderComponent({
-    focusActivationMode: 'manual',
-    onTabSelected: mockOnTabSelected,
-  });
+it('should handle keypresses when focusActivationMode is manual', async () => {
+  const mockonActivated = jest.fn();
+
+  const { container } = renderComponent(
+    {
+      focusActivationMode: 'manual',
+      onValueChange: mockonActivated,
+      defaultValue: 'tab0',
+    },
+    {},
+    <>
+      <Tabs.TabList>
+        <Tabs.Tab value='tab0' key={0} label='Label 0' />
+        <Tabs.Tab value='tab1' key={1} label='Label 1' />
+        <Tabs.Tab value='tab2' key={2} label='Label 2' />
+      </Tabs.TabList>
+
+      <Tabs.Panel value='tab0'>Test Content 0</Tabs.Panel>
+      <Tabs.Panel value='tab1'>Test Content 1</Tabs.Panel>
+      <Tabs.Panel value='tab2'>Test Content 2</Tabs.Panel>
+    </>,
+  );
 
   const tablist = container.querySelector('.iui-tabs') as HTMLElement;
   const tabs = Array.from(tablist.querySelectorAll('.iui-tab'));
 
   // 0 -> 1
-  fireEvent.keyDown(tablist, { key: 'ArrowRight' });
-  expect(mockOnTabSelected).not.toBeCalled();
+  fireEvent.keyDown(tabs[0], { key: 'ArrowRight' });
+  expect(mockonActivated).not.toBeCalled();
   expect(document.activeElement).toBe(tabs[1]);
 
   // select 1
-  fireEvent.keyDown(tablist, { key: 'Enter' });
-  expect(mockOnTabSelected).toBeCalledWith(1);
+  fireEvent.keyDown(tabs[1], { key: 'Enter' });
```

Comment:
Would appreciate some help with this test. I don't know why jest is not clicking enter/activating the tab - it works in dev server. Maybe something in tests?

Reply:
Fixed.